File: intermediate control work on the specialization block the first/notesApplication.py
import json
import datetime
import os

def create_note():
    note_id = input("Введите идентификатор заметки: ")
    title = input("Введите заголовок заметки: ")
    body = input("Введите текст заметки: ")
    created_at = datetime.datetime.now().strftime("%Y-%m-%d %H:%M:%S")
    return {"id": note_id, "title": title, "body": body, "created_at": created_at}

def save_note(note, path):
    filename = os.path.join(path, "notes.json")
    with open(filename, "a", encoding="utf-8") as file:
        json.dump(note, file, ensure_ascii=False)
        file.write("\n")

def read_notes(path):
    filename = os.path.join(path, "notes.json")
    with open(filename, "r", encoding="utf-8") as file:
        notes = file.readlines()
        for note in notes:
            print(note)

def edit_note(path):
    note_id = input("Введите идентификатор заметки для редактирования: ")
    new_body = input("Введите новый текст заметки: ")
    filename = os.path.join(path, "notes.json")
    with open(filename, "r+", encoding="utf-8") as file:
        notes = file.readlines()
        file.seek(0)
        for note in notes:
            note_data = json.loads(note)
            if note_data["id"] == note_id:
                note_data["body"] = new_body
            json.dump(note_data, file, ensure_ascii=False)
            file.write("\n")
        file.truncate()

def delete_note(path):
    note_id = input("Введите идентификатор заметки для удаления: ")
    filename = os.path.join(path, "notes.json")
    with open(filename, "r+", encoding="utf-8") as file:
        notes = file.readlines()
        file.seek(0)
        for note in notes:
            note_data = json.loads(note)
            if note_data["id"] != note_id:
                json.dump(note_data, file, ensure_ascii=False)
                file.write("\n")
        file.truncate()

def main():
    path = r"C:\Users\Maria\OneDrive\GB\intermediate control work on the specialization block\the first"
    while True:
        print("1. Создать заметку")
        print("2. Просмотреть список заметок")
        print("3. Редактировать заметку")
        print("4. Удалить заметку")
        print("5. Выйти")
        choice = input("Выберите действие: ")

        if choice == "1":
            note = create_note()
            save_note(note, path)
            print("Заметка сохранена.")
        elif choice == "2":
            read_notes(path)
        elif choice == "3":
            edit_note(path)
            print("Заметка отредактирована.")
        elif choice == "4":
            delete_note(path)
            print("Заметка удалена.")
        elif choice == "5":
            break
        else:
            print("Некорректный выбор. Попробуйте снова.")

if __name__ == "__main__":
    main()

File: intermediate control work on the specialization block the first/notesApplication.java
import java.io.*;
import java.time.LocalDateTime;
import java.time.format.DateTimeFormatter;
import java.util.Scanner;

public class notesApplication {
    private static final String FILE_PATH = "C:\\Users\\Maria\\OneDrive\\GB\\intermediate control work on the specialization block\\the first";

    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);

        while (true) {
            System.out.println("1. Создать заметку");
            System.out.println("2. Просмотреть список заметок");
            System.out.println("3. Редактировать заметку");
            System.out.println("4. Удалить заметку");
            System.out.println("5. Выйти");
            System.out.print("Выберите действие: ");
            int choice = scanner.nextInt();
            scanner.nextLine(); // Считываем символ новой строки после ввода числа

            switch (choice) {
                case 1:
                    createNote();
                    break;
                case 2:
                    readNotes();
                    break;
                case 3:
                    editNote();
                    break;
                case 4:
                    deleteNote();
                    break;
                case 5:
                    return;
                default:
                    System.out.println("Некорректный выбор. Попробуйте снова.");
            }
        }
    }

    private static void createNote() {
        Scanner scanner = new Scanner(System.in);

        System.out.print("Введите идентификатор заметки: ");
        String noteId = scanner.nextLine();

        System.out.print("Введите заголовок заметки: ");
        String title = scanner.nextLine();

        System.out.print("Введите текст заметки: ");
        String body = scanner.nextLine();

        LocalDateTime now = LocalDateTime.now();
        DateTimeFormatter formatter = DateTimeFormatter.ofPattern("yyyy-MM-dd HH:mm:ss");
        String createdAt = now.format(formatter);

        String note = noteId + ";" + title + ";" + body + ";" + createdAt;

        try (FileWriter fileWriter = new FileWriter(FILE_PATH, true);
             BufferedWriter bufferedWriter = new BufferedWriter(fileWriter)) {
            bufferedWriter.write(note);
            bufferedWriter.newLine();
            System.out.println("Заметка сохранена.");
        } catch (IOException e) {
            System.out.println("Ошибка при сохранении заметки: " + e.getMessage());
        }
    }

    private static void readNotes() {
        try (FileReader fileReader = new FileReader(FILE_PATH);
             BufferedReader bufferedReader = new BufferedReader(fileReader)) {
            String line;
            while ((line = bufferedReader.readLine()) != null) {
                System.out.println(line);
            }
        } catch (IOException e) {
            System.out.println("Ошибка при чтении заметок: " + e.getMessage());
        }
    }

    private static void editNote() {
        Scanner scanner = new Scanner(System.in);

        System.out.print("Введите идентификатор заметки для редактирования: ");
        String noteId = scanner.nextLine();

        System.out.print("Введите новый текст заметки: ");
        String newBody = scanner.nextLine();

        try (FileReader fileReader = new FileReader(FILE_PATH);
             BufferedReader bufferedReader = new BufferedReader(fileReader);
             FileWriter fileWriter = new FileWriter(FILE_PATH + ".tmp");
             BufferedWriter bufferedWriter = new BufferedWriter(fileWriter)) {
            String line;
            while ((line = bufferedReader.readLine()) != null) {
                String[] noteData = line.split(";");
                if (noteData[0].equals(noteId)) {
                    noteData[2] = newBody;
                    line = String.join(";", noteData);
                }
                bufferedWriter.write(line);
                bufferedWriter.newLine();
            }
        } catch (IOException e) {
            System.out.println("Ошибка при редактировании заметки: " + e.getMessage());
            return;
        }

        File file = new File(FILE_PATH);
        File tempFile = new File(FILE_PATH + ".tmp");
        if (file.delete() && tempFile.renameTo(file)) {
            System.out.println("Заметка отредактирована.");
        } else {
            System.out.println("Ошибка при редактировании заметки.");
        }
    }

    private static void deleteNote() {
        Scanner scanner = new Scanner(System.in);

        System.out.print("Введите идентификатор заметки для удаления: ");
        String noteId = scanner.nextLine();

        try (FileReader fileReader = new FileReader(FILE_PATH);
             BufferedReader bufferedReader = new BufferedReader(fileReader);
             FileWriter fileWriter = new FileWriter(FILE_PATH + ".tmp");
             BufferedWriter bufferedWriter = new BufferedWriter(fileWriter)) {
            String line;
            while ((line = bufferedReader.readLine()) != null) {
                String[] noteData = line.split(";");
                if (!noteData[0].equals(noteId)) {
                    bufferedWriter.write(line);
                    bufferedWriter.newLine();
                }
            }
        } catch (IOException e) {
            System.out.println("Ошибка при удалении заметки: " + e.getMessage());
            return;
        }

        File file = new File(FILE_PATH);
        File tempFile = new File(FILE_PATH + ".tmp");
            if (file.delete() && tempFile.renameTo(file)) {
                System.out.println("Заметка удалена.");
            } else {
                System.out.println("Ошибка при удалении заметки.");
            }
        }
    }
                   